modal/_functions.py

```diff
@@ -1865,6 +1902,47 @@ async def get(self, timeout: Optional[float] = None, *, index: int = 0) -> Retur
         """
         return await self._invocation().poll_function(timeout=timeout, index=index)
 
+    @live_method_gen
+    async def iter(self, *args) -> AsyncIterator[Tuple[int, ReturnType]]:
```

Comment:
FWIW my initialy reaction here was "WTF are the *args?". I think the `range()` signature is pretty fussy. Might prefer an explicit `start=0, end=None` parameterization, but don't feel super strongly.